tox/config.py

```diff
@@ -787,7 +787,8 @@ def __init__(self, config, inipath):
             factors = set(name.split('-'))
             if section in self._cfg or factors <= known_factors:
                 config.envconfigs[name] = \
-                    self.make_envconfig(name, section, reader._subs, config)
+                    self.make_envconfig(name, section, reader._subs, config,
+                                        replace=name in config.envlist)
```

Comment:
**Note to reviewers**: this is the core of the change I wanted to make.  Don't replace environment substitutions unless the environment is in `envlist`.